src/Components/ShareCollectionDialog.tsx

```diff
@@ -67,78 +78,79 @@ export const ShareCollectionDialog: React.FC<
 
   const { sendToast } = useToasts()
 
-  useOnce(async () => {
-    setMode("Updating")
-
+  const handleToggle = async toggleValue => {
     try {
-      await submitMutation({
+      setIsUpdating(true)
+      const result = await submitMutation({
         variables: {
           input: {
-            id: collectionId,
-            name: collectionName, // TODO: Shouldn't be required
-            private: false,
+            id: collection.internalID,
+            private: !toggleValue, // negate the toggle, because the flag is for 'private', not 'public'
           },
         },
         rejectIf: res => {
           return !!res.updateCollection?.responseOrError?.mutationError
         },
       })
-
-      setMode("Ready")
+      setSlug(result.updateCollection?.responseOrError?.collection?.slug)
+      setIsShared(toggleValue)
+      setIsUpdating(false)
```

Comment:
Should `setIsUpdating(false)` be placed in a `finally` block to also set it to false when an error occurs?

Reply:
Good catch, thanks